dev/AppNotifications/AppNotificationContentBuilder/AppNotificationContentBuilder.idl

```diff
@@ -0,0 +1,120 @@
+        // "Reply", or "Archive".In the upcoming toast telemetry dashboard in Dev Center, you will be able to view how frequently
+        // your actions are being clicked.
+        String HintActionId{ get; };
+        ButtonConfig SetHintActionId(String hintActionId);
+
+        // Gets or sets the ID of an existing AppNotificationTextBox in order to have this button display to the right of the input.
+        String TextBoxId{ get; };
+        ButtonConfig SetTextBoxId(String textBoxId);
+
+        // Adds a key(without value) to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+        ButtonConfig AddArgument(String key);
+
+        // Adds a key/value to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+        [default_overload]
+        ButtonConfig AddArgument(String key, String value);
+        ButtonConfig AddArgument(String key, Int32 value);
+        ButtonConfig AddArgument(String key, Double value);
+        ButtonConfig AddArgument(String key, Single value);
+        ButtonConfig AddArgument(String key, Boolean value);
+
+        // Retrieves the XML content of the button.
+        String GetXml();
+    }
+
+    [contract(AppNotificationsContract, 2)]
+    enum AdaptiveImageCrop
+    {
+        Default = 0, // Cropping behavior determined by renderer.
+        None = 1, // Image is not cropped.
+        Circle = 2, // Image is cropped to a circle shape.
+    };
+
+    [contract(AppNotificationsContract, 2)]
+    runtimeclass ImageConfig
+    {
+        ImageConfig(Windows.Foundation.Uri imageUri);
+
+        // A description of the image, for users of assistive technologies.
+        String AlternateText{ get; };
+        ImageConfig SetAlternateText(String alternateText);
+
+        // An enum to describe how the image will be cropped.
+        AdaptiveImageCrop HintCrop{ get; };
+        ImageConfig SetHintCrop(AdaptiveImageCrop hintCrop);
+
+        // Override the app logo with custom image of choice that will be displayed on the AppNotification, otherwise it will be an inline image.
+        Boolean IsAppLogo{ get; };
+        ImageConfig SetIsAppLogo(Boolean isAppLogo);
+
+        // A value whether Windows is allowed to append a query string to the image URI supplied in the Tile notification.
+        Boolean AddImageQuery{ get; };
+        ImageConfig SetAddImageQuery(Boolean addImageQuery);
+
+        // Retrieves the XML content of the image.
+        String GetXml();
+    }
+
+    [contract(AppNotificationsContract, 2)]
+    runtimeclass AppNotificationContentBuilder
+    {
+        AppNotificationContentBuilder();
+
+        // Adds a key(without value) to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+        AppNotificationContentBuilder AddArgument(String key);
+
```

Comment:
Same question as for the ButtonConfig API. Do we know of a situation that would require this?